blaze-client/src/main/scala/org/http4s/client/blaze/Http1Connection.scala

```diff
@@ -383,4 +389,7 @@ private object Http1Connection {
       writer
     } else writer
   }
+
+  private val NoPayloadMethods: Set[Method] =
+    Set(Method.GET, Method.DELETE, Method.CONNECT, Method.TRACE)
```

Comment:
This information might be encoded on the Method itself.  (We have a `NoBody` type that's on the way out, but perhaps not even used?  And there's a difference between "MUST not" and "undefined semantics").

Anyway, this does the right thing here.

Reply:
Do you think we should instead just omit the `Content-Length` header if there are no chunks to be written (the observed behavior is an implementation detail of `CachingChunkWriter`)? It might be unintuitive to users who try to submit a body with a `GET` and don't see it written. IIRC this is a poorly defined part of the spec anyways

If we really want to restrict users from passing bodies to methods which don't "support" it, doing that with type-level constraints seems like a more appropriate solution

Reply:
I think a caveat here might be that, we would want to send the `Content-Length: 0` header for POST requests with empty bodies but not for GET requests